src/hal/drivers/mesa-hostmot2/hm2_rpspi.c

```diff
@@ -0,0 +1,568 @@
+	BCM2835_SPICS |= SPI_CS_TA|SPI_CS_CLEAR_TX|SPI_CS_CLEAR_RX;
+
+	/*rtapi_print_msg(RTAPI_MSG_ERR, "Out-Data: ");
+	for (i=0; i<msgsize; i++) {
+		rtapi_print_msg(RTAPI_MSG_ERR, "%08x:", this->txBuf[i]);
+	} 
+		rtapi_print_msg(RTAPI_MSG_ERR, "1 BCM2835_SPICS: %08x:", BCM2835_SPICS);
+	*/
+	
+	tbuff = (uint8_t *)this->txBuf;
+	rbuff = (uint8_t *)this->rxBuf;
+	
+	
+	/* send txBuf */
+	/* RPi SPI transfers 8 bits at a time only */
+	for (i=0; i<msgsize+1; i++) {
+		for(j=3; j>=0; j--){
+			//while(!(BCM2835_SPICS & SPI_CS_TXD));	// wait for  free space in fifo
+			BCM2835_SPIFIFO = tbuff[j];
+			if(!(BCM2835_SPICS & SPI_CS_TXD))
+				rtapi_print_msg(RTAPI_MSG_ERR, "TX-FIFO is full!\n");
+		}
+		
+		/* wait until transfer is finished */
+		while (!(BCM2835_SPICS & SPI_CS_DONE));
+		
+		//rtapi_print_msg(RTAPI_MSG_ERR, "2 BCM2835_SPICS: %08x:", BCM2835_SPICS);
+		
+		if((BCM2835_SPICS & SPI_CS_RXR) || (BCM2835_SPICS & SPI_CS_RXF))
+			rtapi_print_msg(RTAPI_MSG_ERR, "RX-FIFO is full!\n");
+		
+		for(j=3; j>=0; j--){
+			//while(!(BCM2835_SPICS & SPI_CS_RXD));	// wait for data in fifo
+			if(!(BCM2835_SPICS & SPI_CS_RXD))
+				rtapi_print_msg(RTAPI_MSG_ERR, "RX-FIFO is empty!\n");
+			rbuff[j] = BCM2835_SPIFIFO;
+		}
+		tbuff += 4;
+		rbuff += 4;
+	}
+	
+	/*rtapi_print_msg(RTAPI_MSG_ERR, "In-Data: ");
+	for (i=0; i<msgsize; i++) {
+		rtapi_print_msg(RTAPI_MSG_ERR, "%08x:", this->rxBuf[i]);
+	}
+	rtapi_print_msg(RTAPI_MSG_ERR, "\n"); */
+	
+	/* Stop transfer */
+	BCM2835_SPICS &= ~SPI_CS_TA;
+	
+	//rtapi_print_msg(RTAPI_MSG_ERR, "3 BCM2835_SPICS: %08x:", BCM2835_SPICS);
+	
+	/*buff = (char *)this->rxBuf;
+	buff += 4; //skip the cookie */
+	for (i=0; i<msgsize; i++) {
+		buffer32[i] = this->rxBuf[i+1];
+	}
+	
+    return 1;
+}
+
+/*************************************************/
+static int check_cookie(hm2_rpspi_t *board) {
+    uint32_t cookie[4];
+    uint32_t xcookie[4] = {0x55aacafe, 0x54534f48, 0x32544f4d, 0x00000400};
+    int r = hm2_rpspi_read(&board->llio, 0x100, cookie, 16);
+    if(r < 0) return -errno;
+
+    if(memcmp(cookie, xcookie, sizeof(cookie))) {
+        rtapi_print_msg(RTAPI_MSG_ERR, "Invalid cookie\n");
+        rtapi_print_msg(RTAPI_MSG_ERR, "Read: %08x %08x %08x %08x\n",
+            cookie[0], cookie[1], cookie[2], cookie[3]);
+        return -ENODEV;
+    }
+    return 0;
+}
+
+/*************************************************/
+static int read_ident(hm2_rpspi_t *board, char *ident) {
+    return hm2_rpspi_read(&board->llio, 0x40c, ident, 8);
+}
+
+/*************************************************/
+static int probe_board(hm2_rpspi_t *board, uint16_t spiclkdiv) {
+    //printf("probe %d\n", spiclkdiv);
+   
+    board->spiclkdiv = spiclkdiv;
+    board->spibpf = 32;
+
+    int ret = check_cookie(board);
+    if(ret < 0) return ret;
+
+    char ident[8];
+    ret = read_ident(board, ident);
+    if(ret < 0) return ret;
+
+    char *base;
+
+    if(!memcmp(ident, "MESA7I43", 8)) {
```

Comment:
7i43 support would require the driver to be able to download the bitfile to the FPGA at every startup and at present this is an empty function.  (the 7i90 is one-time flashed with mesaflash).  I see no harm in the interfaces being present but undocumented at this time.